org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+                        if ( hsPersonIds.Contains( person.Id ) ) { continue; } // redundant
+
+                        List<HubspotPropertyUpdate> properties = new List<HubspotPropertyUpdate>();
+                        properties.Add( new HubspotPropertyUpdate() { property = "firstname", value = person.NickName } );
+                        properties.Add( new HubspotPropertyUpdate() { property = "lastname", value = person.LastName } );
+                        properties.Add( new HubspotPropertyUpdate() { property = "rock_person_id", value = person.Id.ToString() } );
+                        PhoneNumber mobile = person.PhoneNumbers.FirstOrDefault( n => n.NumberTypeValueId == 12 );
+                        if ( mobile != null && !mobile.IsUnlisted && mobile.IsMessagingEnabled )
+                        {
+                            properties.Add( new HubspotPropertyUpdate() { property = "phone", value = mobile.NumberFormatted } );
+                        }
+
+                        string email = person.Email.ToLower();
+                        string[] emailSep = email.Split( '.' );
+                        string emailtld = "." + emailSep[emailSep.Length - 1];
+                        if ( person.CanReceiveEmail( true ) && hsPersonEmails.Contains( person.Email.ToStringSafe().ToLower() ) == false && email.Contains( "@" ) && tlds.Contains( emailtld ) )
+                        {
+                            properties.Add( new HubspotPropertyUpdate() { property = "email", value = email } );
+                        }
+
+                        var url = $"https://api.hubapi.com/crm/v3/objects/contacts/";
+                        AddHsContact( url, properties, 0 );
+                        idAdded++;
+                    }
+                }
+            }
+
+            // Output Job Stats
+            var resultMsg = new StringBuilder();
+            resultMsg.AppendFormat( "{0} contacts with Rock person id added or updated", idSet );
+            resultMsg.AppendFormat( ", {0} contacts with Rock person id removed", idRemoved );
+            resultMsg.AppendFormat( ", {0} contacts with Rock person id that is valid", idCorrect );
+            resultMsg.AppendFormat( ", {0} contacts with no match in Rock", idEmpty );
+            resultMsg.AppendFormat( ", {0} new contacts imported into HubSpot", idAdded );
+            context.UpdateLastStatusMessage(resultMsg.ToString());
+
+            // Write excel file
+            byte[] sheetbytes = excel.GetAsByteArray();
+            string path = AppDomain.CurrentDomain.BaseDirectory + "\\Plugins\\org_thecrossingchurch\\Assets\\Generated\\" + dataMap.GetString("PotentialMatchesFileName") + ".xlsx";
+            System.IO.FileInfo file = new System.IO.FileInfo(path);
+            file.Directory.Create();
+            System.IO.File.WriteAllBytes(path, sheetbytes);
```

Comment:
Can we add a using statement for System.IO so that we don't have to spell it out each time? That'd help to improve readability.